File: app/src/main/java/com/omatt/fdlsandbox/utils/AppController.java
package com.omatt.fdlsandbox.utils;

import android.app.Application;
import android.content.Context;

import com.omatt.fdlsandbox.components.AppComponent;
import com.omatt.fdlsandbox.components.DaggerAppComponent;

/**
 * Created by omarmatthew on 6/24/17.
 */

public class AppController extends Application{
    private static AppController sInstance;
    private AppComponent appComponent;

    /**
     * Get application instance
     *
     * @return AppController
     */
    public static AppController getsInstance() {
        return sInstance;
    }

    public static AppComponent getComponent(Context context) {
        return ((AppController) context.getApplicationContext()).appComponent;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        sInstance = this;

        // Initialize Dagger AppComponent
        appComponent = DaggerAppComponent.builder().build();
    }
}

File: fdlsandbox-android-kotlin/app/src/main/java/com/omatt/fdlsandbox/AppController.kt
package com.omatt.fdlsandbox

import android.app.Application
import android.util.Log
import com.crashlytics.android.core.CrashlyticsCore
import com.omatt.fdlsandbox.components.AppComponent
import com.omatt.fdlsandbox.components.AppModule
import com.omatt.fdlsandbox.components.DaggerAppComponent
import io.fabric.sdk.android.Fabric

// Suppress appModule deprecation warning
@Suppress("DEPRECATION")
/**
 * Created by omarmatthew on 9/16/2017.
 * App Controller
 */
class AppController : Application() {
    val build = false
    val TAG = "AppController"

    companion object {
        //platformStatic allow access it from java code
        @JvmStatic
        lateinit var component: AppComponent
        @JvmStatic
        var instance: AppController? = null
    }

    override fun onCreate() {
        instance = this
        component = DaggerAppComponent.builder().appModule(AppModule(this))
                .build()

        component.inject(this)

        val crashlyticsCore = CrashlyticsCore.Builder()
                .disabled(true)
                .build()
        Log.i(TAG, "Debug? ${BuildConfig.DEBUG}")
        Fabric.with(this, crashlyticsCore)

        super.onCreate()
    }
}